File: frontend/src/types/paper.ts
// Paper types matching backend models
export interface Paper {
  paperId: string;
  title: string;
  authors: Array<{ name: string; id?: string }>;
  year?: number;
  abstract?: string | null;
  doi?: string | null;
  citationCount?: number;
  url?: string | null;
  venue?: string | null;
  tldr?: string | null;
  source?: string;
}

export interface SearchParams {
  topics?: string;  // comma-separated
  authors?: string; // comma-separated
  sortBy?: 'recency' | 'relevance';
  page?: number;
}

export interface SearchResponse {
  papers: Paper[];
  total: number;
  has_more: boolean;
}

File: backend/models/paper.py
"""
Pydantic models for Paper entities
"""
from pydantic import BaseModel, Field
from typing import List, Optional


class Author(BaseModel):
    """Author information"""
    name: str
    id: Optional[str] = None


class Paper(BaseModel):
    """Research paper model"""
    paper_id: str = Field(..., alias="paperId")
    title: str
    abstract: Optional[str] = None
    authors: List[Author] = []
    year: Optional[int] = None
    venue: Optional[str] = None
    citation_count: int = Field(0, alias="citationCount")
    url: Optional[str] = None
    tldr: Optional[str] = None
    source: str = "openalex"
    doi: Optional[str] = None
    
    class Config:
        populate_by_name = True  # Allow both paperId and paper_id


class PaperSearchParams(BaseModel):
    """Parameters for searching papers"""
    topics: Optional[str] = None
    authors: Optional[str] = None
    sort_by: str = "recency"
    page: int = Field(1, ge=1)
    per_page: int = Field(200, ge=1, le=200)
